pennylane_lightning/lightning_gpu/_state_vector.py

```diff
@@ -248,6 +239,17 @@ def _apply_state_vector(self, state, device_wires, use_async=False):
             ravelled_indices, state, use_async
         )  # this operation on device
 
+    def _select_setBasisState(self, state, wires, use_async):
+        """Select the proper setBasisState from the C++ backend"""
+        if self._mpi_handler.use_mpi:
+            basis_states = 2 ** (self.num_wires - 1 - np.array(list(wires)))
```

Comment:
Any reason to process it in the python layer instead of C++ layer?

Reply:
It took me a while to do it but it is done!!! :rocket:

Reply:
Awsome @LuisAlfredoNu !